### ISSUE TITLE
Support for apex domains in azure_frontdoor

### ISSUE THREAD

**Issue by akitson-degreed** [hashicorp/terraform-provider-azurerm#24787]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

I cannot find a way to add an apex domain to frontdoor.

It looks like there was an old ticket opened for this [here](https://github.com/hashicorp/terraform-provider-azurerm/issues/6781) but even though it states it was resolved, I do not see how.

According to the [MS Documentation](https://learn.microsoft.com/en-us/azure/frontdoor/front-door-how-to-onboard-apex-domain?pivots=front-door-standard-premium), you choose either "custom domain" or "apex domain" when adding a domain to frontdoor (in the portal).  However, there does not seem to be any terraform resource for managing apex domains in frontdoor.  I can only find [azurerm_cdn_frontdoor_custom_domain](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_frontdoor_custom_domain).

If I use the [azurerm_cdn_frontdoor_custom_domain](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_frontdoor_custom_domain) resource to create an apex domain as a custom domain, and then use [azurerm_dns_a_record](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/dns_a_record) to manage the dns record, the domain is created in frontdoor as a custom domain (expected), and the dns "A" record is created (expected), but frontdoor is still looking for a CNAME record and shows an error on the domain screen in the portal with the message "Create CNAME Record".
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/84340121/3730f094-d1af-4530-81ca-86829e90508c)

I would expect there to be an `azurerm_cdn_frontdoor_apex_domain` resource for managing these apex domains.

According to the [MS Documentation](https://learn.microsoft.com/en-us/azure/frontdoor/front-door-how-to-onboard-apex-domain?pivots=front-door-standard-premium), an apex domain properly created in frontdoor will still use "the DNS state column for CNAME mapping check. Since an apex domain doesn’t support a CNAME record, the DNS state will show 'CNAME record is currently not detected' even after you add the alias record to the DNS provider."  However, as shown in the image on the MS Documentation, and copied here, that message is "green" and does not show as an error.
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/84340121/1a2be27a-82cc-4585-b3f5-da69e83adf0d)

I'd like to manage the apex domains for frontdoor correctly in terraform.  Is there another resource I can use for this, or does an `azurerm_cdn_frontdoor_apex_domain` resource need to be created?

### New or Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_apex_domain

**Comment by rcskosir** [comment]
@akitson-degreed Perhaps you already saw this while looking at old related issues, but sharing in case you missed it: https://github.com/hashicorp/terraform-provider-azurerm/issues/8848#issuecomment-820200842

> Azure created a workaround for apex domains: You have to create a regular CNAME record in your DNS, mapping afdverify.<yourdomain> to afdverify.<your front door name>.azurefd.net and then you can deploy Azure Front Door with a frontendEndpoint listening on that apex domain. This is documented at https://docs.microsoft.com/en-us/azure/frontdoor/front-door-custom-domain#map-the-temporary-afdverify-subdomain

**Comment by akitson-degreed** [comment]
@rcskosir yes, I saw that.  But that isn't the issue here.  It isn't about the DNS management.. it's about registering the domain as an "apex" domain in azure frontdoor.

With terraform, I can create the domain in frontdoor as a _custom domain_, but not as an _apex domain_.  There is no terraform resource for creating an apex domain in azure frontdoor.  As you can see in the image below, when adding a domain to azure frontdoor using the portal, there is an option to add it as an apex domain.  But there is no terraform resource to do that.  The only terraform resource for adding a domain to azure frontdoor is [azurerm_cdn_frontdoor_custom_domain](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_frontdoor_custom_domain). 
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/84340121/23e54e4a-368f-46e0-a655-adfc14ef8331)

**Comment by rcskosir** [comment]
@akitson-degreed Thank you for the clarification!

**Comment by MattWhite-personal** [comment]
@akitson-degreed did you find a fix or workaround for this - seems to have been open for a while and just noticing this with testing some resources

**Comment by MattWhite-personal** [comment]
@rcskosir is there a way to support looking at this resource against the apex domain - I would have a look at the code but never developed against terraform before. Is this a limitation in what Microsoft exposes via their API rather than what Terraform can currently support as a resource

**Comment by MattWhite-personal** [comment]
I've realised that this is actually possible so long as you remove the Azure DNS zone from the terraform configuration. Example code below

``` terraform
resource "azurerm_cdn_frontdoor_endpoint" "static-web-app" {
  name                     = "afd-ep-swa-test"
  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.test.id
}

resource "azurerm_cdn_frontdoor_custom_domain" "static-web-app" {
  name                     = "afd-cd-swa-test"
  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.test.id
  host_name                = azurerm_dns_zone.tftest.name

  tls {
    certificate_type = "ManagedCertificate"
  }
}

resource "azurerm_dns_a_record" "static-web-app" {
  name                = "@"
  zone_name           = azurerm_dns_zone.tftest.name
  resource_group_name = azurerm_resource_group.dnszones.name
  ttl                 = 300
  target_resource_id  = azurerm_cdn_frontdoor_endpoint.static-web-app.id
}

resource "azurerm_dns_txt_record" "dnsauth" {
  name                = "_dnsauth"
  zone_name           = azurerm_dns_zone.tftest.name
  resource_group_name = azurerm_resource_group.dnszones.name
  ttl                 = 300

  record {
    value = azurerm_cdn_frontdoor_custom_domain.static-web-app.validation_token
  }
}
```

Just need to allow time post resource creation for the verification to take place but this is likely normal with a non-apex domain setup
